scipy/stats/stats.py

```diff
@@ -1509,16 +1527,29 @@ def skewtest(a, axis=0, nan_policy='propagate'):
         Defines how to handle when input contains nan.
         The following options are available (default is 'propagate'):
 
-          * 'propagate': returns nan
-          * 'raise': throws an error
-          * 'omit': performs the calculations ignoring nan values
+        * 'propagate': returns nan
+        * 'raise': throws an error
+        * 'omit': performs the calculations ignoring nan values
+
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        The following options are available:
+
+        * 'two-sided': the skewness of the distribution underlying the sample
```

Comment:
I think that explicitly stating that this is the alternative hypothesis is better here. 

```suggestion
        * 'two-sided': The alternative hypothesis is that the skewness of the distribution underlying the sample
```

From the view of a new user, this makes it easier to understand whether the statement is the null hypothesis or the alternative hypothesis. Also, IIRC, other tests that you have documented do this. Do you think doing this would be helpful here too?

Reply:
> Also, IIRC, other tests that you have documented do this. 

Do you mean gh-13188? I was just pushing buttons for others there. The reason why it says "the alternative is..." every time is that it first says "the null hypothesis is ..." for each option. But in that PR, I don't understand how you have different null hypotheses corresponding with the different alternatives. For instance, I understand

> The null hypothesis is that F(x) = G(x) for all x

But how do you calculate the distribution of the test statistic under

> The null hypothesis is that F(x) >= G(x) for all x

? Is that really what's done? 

In every statistical test I've thought much about, there is just one null hypothesis corresponding for all three alternative hypotheses. The thing that changes is how you calculate the p-value given the distribution of the test statistic under the assumption of the null hypothesis.

---

Back to this PR. If we're not stating the null hypothesis, I don't think it needs to be stated so many times that this is the alternative hypothesis. I wanted to remove that redundancy. 

I thought about moving it above like:
> Defines the alternative hypothesis. Default is 'two-sided'.
   The following options *for the alternative hypothesis* are available:

But that's still redundant. The parameter is called "alternative", and we just said "Defines the alternative hypothesis", so I would be surprised if someone still wondered whether these were null hypotheses or alternative hypotheses. We said it was the alternative twice already.

On the other hand, I do agree with the idea that we should state the null hypothesis. So if we were to do that, I might write:

> Defines the alternative hypothesis. Default is 'two-sided'.

> The null hypothesis is that the skewness of the distribution underlying the sample is the same as that of the normal 
   distribution. The following options for the alternative hypothesis are available:

Would that be better? 
Is that accurate? Is it true that the null hypothesis is the same for all three cases?

Reply:
Never mind. The null is stated at the top. I say leave it there.

Reply:
> Do you mean gh-13188?

Yes, I was talking about that PR.

> Would that be better?

I think it would be OK both ways: writing null and alternative hypotheses or just writing the alternative hypothesis. For the sake of completeness, it would be better to write both the null and alternate hypotheses.

> Is that accurate? Is it true that the null hypothesis is the same for all three cases?

I don't think so. For instance, when `alternative='less'` for the skew test, the null hypothesis is that the skewness of the sample is greater than that of the normal distribution (i. e. > 0). It also changes the same way for when `alternative='greater'`. If this sounds like too much work for this PR, we can leave this for some other time. What do you say?

Reply:
> Do you still think these might be confused with null hypotheses?

I don't think it engenders confusion. I thought that being explicit would just make it easier to understand. But from your comment earlier:

> But that's still redundant. The parameter is called "alternative", and we just said "Defines the alternative hypothesis", so I would be surprised if someone still wondered whether these were null hypotheses or alternative hypotheses. We said it was the alternative twice already.

I think it's OK not to introduce redundancy.

Reply:
OK. Thanks for the rapid feedback! So much easier to get things done in near real-time!

Reply:
But

> I don't think so. For instance, when alternative='less' for the skew test, the null hypothesis is that the skewness of the sample is greater than that of the normal distribution (i. e. > 0)

We're back to something I don't understand, as I wrote [above regarding kstest](https://github.com/scipy/scipy/pull/13549#discussion_r594923594). If this is the case, then the null hypothesis at the top of the documentation is not correct for all three alternatives and would need to be changed.

Reply:
Consider, for example, [the binomial test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom_test.html).

The null hypothesis for all three alternatives is the same: that the probability of "success" is the provided `p`. The distribution of the test statstic (the obverved number of successes) is calculated under that same assumption for all alternatives. The alternatives just control which tail you look at. Right?

How are these other tests different?

Reply:
> Consider, for example, [the binomial test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom_test.html).
> 
> The null hypothesis for all three alternatives is the same: that the probability of "success" is equal to the provided `p`. The distribution of the test statstic (the observed number of successes) is calculated under that same null hypotheis for all alternatives - it is the binimial distribution with the provided `n` and `p`. The alternatives just control which tail you look at. Right?
> 
> How are these other tests different?

I think that the null hypothesis is everything other than the alternate hypothesis for normality tests. I will have to see if that still holds true for other tests though. If I am not wrong, we cannot say if the statistic is exactly the null value if we get a high p-value at the left/right tail which is why we formulate the null hypothesis in such a way that covers the rest of the distribution (which is not covered by the alternate hypothesis). Does this make sense? I will try to rephrase or attach some source if it's unclear.

Reply:
> If I am not wrong, we cannot say if the statistic is exactly the null value

Statistical tests never tell us whether the null hypothesis is true. They can only give us evidence for rejecting it. And that distinction is _very_ important to this issue. 

So I'm afraid that doesn't make sense as written. I wrote a longer post explaining, but I'll wait until you reply to post it.

Reply:
> Statistical tests never tell us whether the null hypothesis is true. They can only give us evidence for rejecting it. And that distinction is _very_ important to this issue.

That aligns with my understanding. I wasn't able to write it properly. Sorry!

> So I'm afraid that doesn't make sense as written. I wrote a longer post explaining, but I'll wait until you reply to post it. You might want to read the conversation at [gh-12518](https://github.com/scipy/scipy/issues/12518), as this has been discussed before.

I looked at gh-12518. My understanding about the null hypothesis being equality/weak inequality was wrong. Statsmodels also uses the equality null hypothesis. So, it would make sense to do that here too. Very sorry for the noise here. Please go ahead and post your longer post if you want. Also, thanks for writing a long post! Really appreciate your help here!

Reply:
OK here it is.

---
If failing to reject the null hypothesis meant "The null hypothesis is correct", e.g. "the true value of the statistic is exactly the null value", then I can see why we might have a problem stating the null hypothesis with a precise value. But there are lots of other problems in that alternate universe : )

Here, we 
1. assume a (precise) null hypothesis about the underlying population,
2. derive a distribution of the sample statistic under the assumption of the null hypothesis, and
3. compare the observed value of the statistic against that distribution. 

If the observed value of the statistic is sufficiently extreme when compared to the disribution (where "sufficiently extreme" is defined by our choice of alternative and significance level), then we say we have evidence to reject the null hypothesis in favor of the alternative. When we fail to reject the null hypothesis, we do not confirm that the null hypothesis is true. We just don't have evidence in favor of the alternative.

Regarding the tests in this PR: the only thing that changes in these tests for the different alternatives is the way we compare the observed value of the test statistic against the distribution (which is derived under the null hypothesis). So even from the code we can see that distribution - and thus the null hypothesis itself - doesn't change depending on the choice of alternative.